File: решения/java/src/Lab15/task6876/Shape.java
package Lab15.task6876;

/**
 * @author akryukov
 *         17.07.2017
 */
public abstract class Shape {
    private String stroke;
    private int strokeWidth;
    private String fill;

    public Shape(String stroke, int strokeWidth, String fill) {
        this.stroke = stroke;
        this.strokeWidth = strokeWidth;
        this.fill = fill;
    }

    public String getStroke() {
        return stroke;
    }

    public int getStrokeWidth() {
        return strokeWidth;
    }

    public String getFill() {
        return fill;
    }

    public abstract double area();

    public static String getHeader(){
        return "&lt;svg width=\"150\" height=\"150\"&gt;";
    }

    public static String getFooter(){
        return "&lt;/svg&gt;";
    }
}

File: решения/cs/Classes/task6876/Shape.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Classes.Task6876
{
    public abstract class Shape
    {
        protected String stroke;
        protected int strokeWidth;
        protected String fill;

        public Shape(String stroke, int strokeWidth, String fill)
        {
            this.stroke = stroke;
            this.strokeWidth = strokeWidth;
            this.fill = fill;
        }

        public abstract double Area { get; }

        public static String Header
        {
            get
            {
                return "<svg width=\"150\" height=\"150\">";
            }
        }

        public static String Footer
        {
            get
            {
                return "</svg>";
            }
        }
    }
}
